src/sentry/search/snuba/executors.py

```diff
@@ -564,3 +569,244 @@ def calculate_hits(
+        #         # Temp solution is to set value to -1, which should never be a real release ID.
+        #         converted_filter[2] = -1
+        #     else:
+        #         converted_filter[2] = release[0].id
+
+        return converted_filter
+
+    def query(
+        self,
+        projects,
+        retention_window_start,
+        group_queryset,
+        environments,
+        sort_by,
+        limit,
+        cursor,
+        count_hits,
+        paginator_options,
+        search_filters,
+        date_from,
+        date_to,
+    ):
+
+        now = timezone.now()
+        end = None
+        end_params = filter(None, [date_to, get_search_filter(search_filters, "date", "<")])
+        if end_params:
+            end = min(end_params)
+
+        if not end:
+            end = now + ALLOWED_FUTURE_DELTA
+
+        retention_date = max(filter(None, [retention_window_start, now - timedelta(days=90)]))
+
+        start_params = [date_from, retention_date, get_search_filter(search_filters, "date", ">")]
+        start = max(filter(None, start_params))
+
+        end = max([retention_date, end])
+
+        if start == retention_date and end == retention_date:
+            return self.empty_result
+
+        if start >= end:
+            return self.empty_result
+
+        for i in range(len(search_filters)):
+            search_filter = search_filters[i]
+            if search_filter.key.name == "first_release":
+                # The filter's value will be the release's "version". Snuba only knows about ID. So we convert version to id here.
+                release = Release.objects.filter(
+                    version=search_filter[2].value, organization_id=projects[0].organization_id
+                )
+                if not release:
+                    return self.empty_result
+                else:
+                    search_filters[i] = SearchFilter(
+                        SearchKey("first_release"), "=", SearchValue(release[0].id)
+                    )
+
+        sort_field = self.sort_strategies[sort_by]
+        chunk_growth = options.get("snuba.search.chunk-growth-rate")
+        max_chunk_size = options.get("snuba.search.max-chunk-size")
+        chunk_limit = limit
+        offset = 0
+        num_chunks = 0
+        hits = self.calculate_hits(
+            sort_field,
+            projects,
+            retention_window_start,
+            group_queryset,
+            environments,
+            sort_by,
+            limit,
+            cursor,
+            count_hits,
+            paginator_options,
+            search_filters,
+            start,
+            end,
+        )
+        if hits == 0:
+            return self.empty_result
+
+        paginator_results = self.empty_result
+        result_groups = []
+        result_group_ids = set()
+
+        max_time = options.get("snuba.search.max-total-chunk-time-seconds")
+        time_start = time.time()
+
+        while (time.time() - time_start) < max_time:
```

Comment:
Not sure I get whether it is supposed to be this class to add the table prefix to every column or you expect that to happen at the callsite. This class is the only one that knows which dataset is querying so it should be the one that adds the table prefix.